sdk/metric/internal/aggtor/aggregator.go

```diff
@@ -0,0 +1,27 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build go1.18
+// +build go1.18
+
+package aggtor // import "go.opentelemetry.io/otel/sdk/metric/internal/aggtor"
+
+// Aggregator forms an aggregation from a collection of recorded measurements.
+type Aggregator[N int64 | float64] interface {
+	// Record includes value in the aggregation.
+	Record(value N)
+
+	// Aggregate returns an aggregation of all recorded values.
+	Aggregate() Aggregation
+}
```

Comment:
I would recommend that Record take a `value` and an `attribute.Set`, and Aggregate() return []Aggregation that also has attributes.  

The reason is if the instrument has to keep track of `Aggregator`s by attributes then it will also need to know how to create new `Aggregator`s when it sees a new attribute set.  This is a process defined by the views that apply and the resulting aggregation selection.

If you make the Aggregation keep track of attributes then you only create `Aggregation`s when you first create the instrument, it's still the same complicated process but it stays contained within the meter.  You will also have the advantage that when you call Aggregate() the []Aggregation does not need to be combined in any way to translate into otlp data.

Reply:
I wen't back and forth on this. I think the goal of matching the otlp is a good one. I'll update the interface to this.